node/core/pvf/src/prepare/worker.rs

```diff
@@ -358,8 +358,15 @@ pub fn worker_entrypoint(socket_path: &str, node_version: Option<&str>) {
 				gum::error!(
 					target: LOG_TARGET,
 					%worker_pid,
-					"Node and worker version mismatch, node needs restarting",
+					"Node and worker version mismatch, node needs restarting, forcing shutdown",
 				);
+				unsafe {
+					// SAFETY: Neither `getppid()` nor `kill()` never fails
+					let ppid = libc::getppid();
+					if ppid > 1 {
+						libc::kill(ppid, libc::SIGKILL);
+					}
+				}
```

Comment:
Can be extracted into a safe function (with the safety guarantee documented) -- to avoid use of `unsafe` here -- in `worker_common.rs`.

Reply:
I like the simplicity of this approach BTW. No need to even send a response, just kill the parent from the child!